src/core/api/border_router_api.cpp

```diff
@@ -37,29 +37,28 @@
 
 #include <openthread/border_router.h>
 
+#include "common/debug.hpp"
 #include "common/instance.hpp"
 #include "common/locator-getters.hpp"
 
 using namespace ot;
 
 otError otBorderRouterGetNetData(otInstance *aInstance, bool aStable, uint8_t *aData, uint8_t *aDataLength)
 {
-    otError   error    = OT_ERROR_NONE;
     Instance &instance = *static_cast<Instance *>(aInstance);
 
-    VerifyOrExit(aData != NULL && aDataLength != NULL, error = OT_ERROR_INVALID_ARGS);
+    assert(aData != NULL && aDataLength != NULL);
```

Comment:
add note in header file to clarify the pointers are not allowed to be NULL

Reply:
Then we must clarify for almost every API, `aInstance` is not allowed to be NULL. Thoughts?